site/src/pages/RoadmapPage/AddYearPopup.tsx

```diff
@@ -13,20 +13,37 @@ const AddYearPopup: FC<AddYearPopupProps> = ({ placeholderYear }) => {
   const dispatch = useAppDispatch();
   const [year, setYear] = useState(placeholderYear);
   const [show, setShow] = useState(false);
-  const target = useRef(null);
+  const target = useRef<HTMLButtonElement>(null);
+  const popupTarget = useRef<HTMLDivElement>(null);
 
   useEffect(() => { setYear(placeholderYear) }, [placeholderYear]);
 
-  const handleClick = (event: React.MouseEvent) => {
-    setShow(!show);
-  };
+  useEffect(() => {
```

Comment:
I'd suggest adding a quick comment here explaining the gist of the change and what it's purpose is, so it's easier to maintain or apply to other areas in the future